ReactWindows/ReactNative.Shared/UIManager/BorderedCanvasManager.cs

```diff
@@ -14,14 +15,52 @@ namespace ReactNative.UIManager
     public abstract class BorderedCanvasManager<TCanvas> : BorderedViewParentManager<TCanvas>
         where TCanvas : Canvas
     {
+        protected override bool HasBorder(TCanvas view)
+        {
+            return GetBorder(view) != null;
+        }
+
+        private Border GetBorder(TCanvas view)
+        {
+            return view.Children[0] as Border;
+        }
+
+        protected override Border GetOrCreateBorder(TCanvas view)
+        {
+            var border = GetBorder(view);
+
+            if (border == null) // check Tag here?
+            {
+                border = new Border { BorderBrush = s_defaultBorderBrush, Tag = 123 };
+                view.Children.Insert(0, border);
+
+                border.SetBinding(FrameworkElement.WidthProperty, new Binding
```

Comment:
Not sure what would be more of a hack, but the ViewManager class has the `SetDimensions` method that you can override to intercept the width/height settings of the view.